### PR TITLE
rafhttp: make TestStreamWriterAttachOutgoingConn more robust

### PR DESCRIPTION
**Pull request by xiang90**
fix https://github.com/coreos/etcd/issues/3983

/cc @heyitsanthony @gyuho

### CODE REVIEW

File: rafthttp/stream_test.go [hunk 1]
```diff
@@ -42,25 +42,39 @@ func TestStreamWriterAttachOutgoingConn(t *testing.T) {
...
 	var wfc *fakeWriteFlushCloser
 	for i := 0; i < 3; i++ {
 		prevwfc := wfc
 		wfc = &fakeWriteFlushCloser{}
 		sw.attach(&outgoingConn{t: streamTypeMessage, Writer: wfc, Flusher: wfc, Closer: wfc})
-		testutil.WaitSchedule()
+
+		// sw.attach happens asynchronously. Waits for its result in a for loop to make the
+		// test more robust on slow CI.
+		for i := 0; i < 3; i++ {
```
Comment (gyuho) [inline rafthttp/stream_test.go]: Maybe use `j` to make different than the one above. It will make it clearer. Thanks.

**Comment by gyuho** [comment]
LGTM.

File: rafthttp/stream_test.go (hunk 1 above)
Comment (xiang90) [inline rafthttp/stream_test.go]: right.

**Comment by xiang90** [comment]
@gyuho Fixed your comment. Thanks for the quick review.

**Pull request merged**